app/Http/Controllers/Auth/RegisteredUserController.php

```diff
@@ -43,13 +56,26 @@ public function store(Request $request): RedirectResponse
             'g-recaptcha-response' => app()->environment('production') ? ['required', new Recaptcha($request->ip())] : [],
         ]);
 
+        $request->session()->forget(['github_email', 'github_username']);
+
         $user = User::create([
             'name' => $request->name,
             'email' => $request->email,
             'username' => $request->username,
             'password' => Hash::make($request->string('password')->value()),
         ]);
 
+        if (is_string($githubUsername) && $githubUsername !== '') {
+            $errors = $gitHub->linkGitHubUser($githubUsername, $user, $request);
+            if ($errors !== []) {
+                $user->delete();
```

Comment:
This will happen if there is already a user linked to the GitHub account but their email does not match the email in GitHub. In this case, we cannot register the user linked to the GitHub account, so we remove the recently created user and redirect them back to the register page so that they can try to register again without linking to GitHub